nucliadb_models/src/nucliadb_models/search.py

```diff
@@ -389,6 +389,26 @@ class SortOptions(BaseModel):
     order: SortOrder = SortOrder.DESC
 
 
+class RankFusionName(str, Enum):
+    LEGACY = "legacy"
+    RECIPROCAL_RANK_FUSION = "rrf"
+
+
+class _BaseRankFusion(BaseModel):
+    name: str
+
+
+class LegacyRankFusion(_BaseRankFusion):
+    name: Literal[RankFusionName.LEGACY] = RankFusionName.LEGACY
+
+
+class ReciprocalRankFusion(_BaseRankFusion):
+    name: Literal[RankFusionName.RECIPROCAL_RANK_FUSION] = RankFusionName.RECIPROCAL_RANK_FUSION
+
+
+RankFusion = Union[LegacyRankFusion, ReciprocalRankFusion]
+
```

Comment:
I think you need to do something similar to what we do with rag strategies so that pydantic can parse/validate the correct model at hand

```python
RagStrategies = Annotated[
    Union[
        FieldExtensionStrategy,
        FullResourceStrategy,
        HierarchyResourceStrategy,
        NeighbouringParagraphsStrategy,
        MetadataExtensionStrategy,
        PreQueriesStrategy,
    ],
    Field(discriminator="name"),
```

Reply:
Done :)